### DISCUSSION TITLE
Rationale behind restrictions on atlas labels and masks?

### DISCUSSION THREAD

**Discussion by jeremie-fouquet** [CoBrALab/RABIES#241] [General]
I'm trying to use the preprocessing pipeline of RABIES in a setting different from the default one, i.e., with rats data and diffusion data.

I provide the SIGMA rat atlas images, masks and labels, but they don't meet RABIES requirements. Namely:
- The WM and CSF labels are not binaries but probabilistic maps. (Sidenote: it is not so uncommon among the available atlases.)
- The SIGMA labels are not sampled the same way as the SIGMA in vivo image (they are in the same space, but the labels resolution is higher than the in vivo image).

How realistic would it be to change these hard constraints on the atlas data to, let's say, warnings?

Thanks.

**Comment by gdevenyi** [comment]
> The SIGMA labels are not sampled the same way as the SIGMA in vivo image (they are in the same space, but the labels resolution is higher than the in vivo image).

This is a known bug that was fixed, see RABIES issues

**Reply by jeremie-fouquet** [reply]
Ok. I see in issue #174 that it was fixed by using antsApplyTransform before RABIES to resample the labels.

I thought RABIES was also applying antsApplyTransform to the labels so that it could handle different resolutions in the atlas files. What exactly is the reason for enforcing the same resolution among all atlas files?

**Reply by gdevenyi** [reply]
It has been a standard for a long time that "an atlas" means an image + a hard segmentation at the same resolution and bounding box.

The SIGMA guy said it was a bug and would fix it, https://github.com/grandjeanlab/MultiRat/issues/1

> What exactly is the reason for enforcing the same resolution among all atlas files?

This is not enforcement, this is assumption of standard practice.

**Reply by jeremie-fouquet** [reply]
Ok, great, I had not found this MultiRat issue. I guess my atlas version was not corrected yet. In this specific case, they had in vivo and ex vivo images with different resolutions with only one set of labels, but I agree it is a rare and uncommon practice. Resampling the labels will do. Thanks.

**Comment by gdevenyi** [comment]
> The WM and CSF labels are not binaries but probabilistic maps. (Sidenote: it is not so uncommon among the available atlases.)

If they are not hard, then they're not labels. Sounds like something from SPM land of "we think we can classify animal tissue"

**Reply by jeremie-fouquet** [reply]
Indeed, label was not exactly the right word choice. Still, probabilistic maps seem to be standard at least for rat atlases. See, for instance, Figure 1 comparing atlases in the SIGMA atlas paper: doi.org/10.1038/s41598-020-63965-x. 

What is the reason for enforcing binary masks?

**Reply by gdevenyi** [reply]
> What is the reason for enforcing binary masks?

Again, not enforcement, requirement, and definition.

Similar to the label discussion, each voxel in an atlas has a specific assignment of membership of a group, WM/CSF. This is needed for similarly hard-cutting the fMRI data, so probabilities aren't useful here.

Probabilities come from the land of human data where 1-1 mapping on the cortex is hard/impossible in volumetric images, but you have nice tissue contrast, and can use these probability ideas to boostrap a intensity-based classification of tissue (something not reasonable to do in Mice/Rats)

**Reply by jeremie-fouquet** [reply]
Ok, thanks for the clarifications. I'm interested in the probabilistic maps for Particle Filtering Tractography (adapting a diffusion pipeline from Maxime Descoteaux to rodent data). It uses GM, WM, and CSF probabilistic maps to reorient fibres that stopped prematurely. 

I think probabilistic maps make some sense in rodents when they try to represent voxels with partial volume.

Anyway, if RABIES needs binary masks, I'll simply apply the transforms (which RABIES appears to save cleverly) to the atlas maps.

**Reply by gdevenyi** [reply]
> I think probabilistic maps make some sense in rodents when they try to represent voxels with partial volume.

Yes definitely makes sense there, where the tissues are unclear, and you're "counting" non-real things like tractography already.
